event/consumer.go

```diff
@@ -60,6 +59,7 @@ func (consumer *Consumer) Consume(
 					<-delay.C
 				}
 
+				ctx = msg.Context()
```

Comment:
```suggestion
				messageCtx := msg.Context()
```

Reply:
yes when I though about it I think the kafka context is only needed when you add a message to batch. I got confused myself between taceid and batchid (which should be there for the entire batch). I have changed it now  so that the kafkacontext is only specific to the message consumed from kafka.